docs/token/non-standard.md

```diff
@@ -0,0 +1,27 @@
+# Non-standard behaviour and gotchas
```

Comment:
And this one; I've tried to document as many leading-to-bugs behaviours in others' code as I could.